homeassistant/config.py

```diff
@@ -357,3 +367,90 @@ def set_time_zone(time_zone_str):
         _LOGGER.warning(
             'Incomplete core config. Auto detected %s',
             ', '.join('{}: {}'.format(key, val) for key, val in discovered))
+
+
+def _log_pkg_error(package, component, config, message):
+    """Log an error while merging."""
+    message = "Package {} setup failed. Component {} {}".format(
+        package, component, message)
+
+    pack_config = config[CONF_PACKAGES].get(package, config)
+    message += " (See {}:{}). ".format(
+        getattr(pack_config, '__config_file__', '?'),
+        getattr(pack_config, '__line__', '?'))
+
+    _LOGGER.error(message)
+
+
+def _identify_config_schema(module):
+    """Extract the schema and identify list or dict based."""
+    try:
+        schema = module.CONFIG_SCHEMA.schema[module.DOMAIN]
+    except (AttributeError, KeyError):
+        return (None, None)
+    t_schema = str(schema)
+    if t_schema.startswith('<function ordered_dict') or \
```

Comment:
You can access the wrapped type at `schema.schema`. You can then use `isinstance(schema.schema, cv.ordered_dict)`